blog/content/post/2025-04-10-CROSSCON-its-Hypervisor-and-Zarhus.md

```diff
@@ -0,0 +1,481 @@
+tmpfs                /var/volatile        tmpfs      defaults              0  0
+
+# uncomment this if your device has a SD/MMC/Transflash slot
+#/dev/mmcblk0p1       /media/card          auto       defaults,sync,noauto  0  0
+
+/dev/mmcblk1p1  /boot   vfat    defaults    0   0
+user in ~ λ
+```
+
+### Issues logging in
+
+I really thought it was the end of weird fixes by then, but there was one final
+one. I was running into problems when the kernel was booting without any errors,
+but suddenly freezing at some point. Initially I expected it to be a login
+issue, so I was looking at the login service and other related things.
+
+This turned out to not be the cause after all - we
+[got info](https://github.com/crosscon/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/issues/8#issuecomment-2702293550)
+that repeatedly printing a message in a specific abort handler within the
+Hypervisor fixes the issue of freezing and not being able to log in.
+
+This is a Hypervisor related issue, and this is just a temporary workaround -
+but it allows us to use the setup with `rootfs` and Zarhus.
+
+### Summary
+
+The result of all this debugging is a
+[ready-to-follow](https://docs.zarhus.com/guides/rpi4-crosscon-hypervisor/)
+guide on how this can be achieved.
+
+It takes the user step-by-step on what changes need to be make in order to
+get this setup to work.
+
+There are still things to be added - right now we are working on recipes
+inside `Yocto`, that will provide us with utilities such as `xtest`, a
+`tee-supplicant` service, and custom drivers that will let us interact with
+the `OPTEE-OS` VM properly. This will be the next big step in integrating
+Zarhus and the CROSSCON Hypervisor together.
+
+---
+
+## Conclusion
+
+The successful port of Zarhus to the CROSSCON hypervisor on the RPi4 will make
+life a lot easier when working with TA's on the Hypervisor, or trying to
+execute security tests, or run any custom program. We hope that this will be
+a big leap forward in flexibility and productivity - quite a big part of the
+CROSSCON project is testing the whole stack, and it will be useful to be able
+to do that straight from the linux VM itself, including compilation and.
+tweaking.
+
+Also, since this is now a `Yocto` based setup, adding any new packages or
+tools should be a breeze.
+
+If you've been using the CROSSCON Hypervisor demo on the RPi and trying to
+test something on the Linux VM, there's a high chance you found the minimal
+initramfs environment limiting. We suggest giving
+[our setup](https://docs.zarhus.com/guides/rpi4-crosscon-hypervisor/) a try -
+we are excited to see how developers might use this, and we remain committed to
+expanding this solution.
+
+For any questions or feedback, feel free to contact us at
+<contact@3mdeb.com> or hop on our community channels:
+
+- [Dasharo Matrix Workspace](https://matrix.to/#/#dasharo:matrix.org)
+- join our quarterly [Dasharo Events related to Zarhus](https://vpub.dasharo.com/o/1)
```

Comment:
Let's use:
https://matrix.to/#/#zarhus:matrix.3mdeb.com

https://vpub.dasharo.com/e/22/zarhus-developers-meetup-0x1

and of course adjust for Zarhus